src/main/java/org/threadly/litesockets/TCPClient.java

```diff
@@ -319,4 +322,81 @@ public boolean isEncrypted() {
     }
     throw new IllegalStateException("Must Set the SSLEngine before starting Encryption!");
   }
+  
+  /**
+   * 
+   * @author lwahlmeier
+   *
+   */
+  private class TCPSocketOptions extends BaseClientOptions {
+    
+    @Override
+    public boolean setTcpNoDelay(boolean enabled) {
+        try {
+          channel.socket().setTcpNoDelay(enabled);
+          return true;
```

Comment:
So I am definitely open to other ideas.  I did the enum thing before, but depreciated it because it seemed to messy.  The true/false just indicates if it was able to set, it gets kind of awkward w/o it because certain socket level option can only be set at certain levels by certain OSes.  mainly the sent/recv buffers.  So true means you set it, false means you didnt and you cant set it at least to the value you you tried to set it at.  Certain things like setting udp options on a tcp connection will just always return false.